### PR TITLE
fix(toHaveStyle): strictly match empty values

### PR DESCRIPTION
**Pull request by just-boris**
**What**:

fixes the issue #272

**Why**:

Current implementation allows false positive matches. More examples in the linked issue 

**Checklist**:

- [ ] n/a Documentation
- [x] Tests
- [ ]  n/a Updated Type Definitions
- [x] Ready to be merged

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -192,12 +192,6 @@ function compareArraysAsSet(a, b) {
   return undefined
 }
 
-function parseJStoCSS(document, css) {
-  const sandboxElement = document.createElement('div')
-  Object.assign(sandboxElement.style, css)
-  return sandboxElement.style.cssText
-}
```
Comment (just-boris) [inline src/utils.js:199]: This is currently not used anywhere else, so I decided to drop this completely

File: src/to-have-style.js [hunk 2]
```diff
@@ -37,7 +38,7 @@ function printoutStyles(styles) {
 // received computed styles
 function expectedDiff(expected, computedStyles) {
   const received = Array.from(computedStyles)
-    .filter(prop => expected[prop])
+    .filter(prop => expected[prop] !== undefined)
```
Comment (just-boris) [inline src/to-have-style.js:41]: This is needed to get a proper diff for `expect(something).toHaveStyle({width: ''})` (expect width to be empty). Without this change it prints "the diff has no visual difference"

File: src/to-have-style.js [hunk 3]
```diff
@@ -21,7 +21,8 @@ function isSubset(styles, computedStyle) {
     !!Object.keys(styles).length &&
     Object.entries(styles).every(
       ([prop, value]) =>
-        computedStyle.getPropertyValue(prop.toLowerCase()) === value,
+        computedStyle[prop] === value ||
+        computedStyle[prop.toLowerCase()] === value,
```
Comment (just-boris) [inline src/to-have-style.js:25]: `computedStyle.getPropertyValue` supports only dash-cased notation. Property getters work for both dash and camelCase

**Comment by gnapse** [comment]
@all-contributors ass @just-boris for bug, code, test

**Comment by gnapse** [comment]
@all-contributors pleas add @just-boris for bug, code, test

**Comment by gnapse** [comment]
@all-contributors please add @just-boris for bug, code, test

**Comment by gnapse** [comment]
@all-contributors please add @just-boris for bug, code, test

**Pull request merged**